File: Application/app/src/main/java/com/example/sage/foodsharebook/models/Ingredient.java
package com.example.sage.foodsharebook.models;

import java.util.List;

import com.google.gson.annotations.SerializedName;


public class Ingredient{

	@SerializedName("image")
	private String image;

	@SerializedName("measures")
	private List<Integer> measures;

	@SerializedName("name")
	private String name;

	@SerializedName("description")
	private String description;

	@SerializedName("id")
	private int id;

	public Ingredient(String name, String description, List<Integer> measures){
		this.name = name;
		this.description = description;
		this.measures = measures;
	}

	public void setImage(String image){
		this.image = image;
	}

	public String getImage(){
		return image;
	}

	public void setMeasures(List<Integer> measures){
		this.measures = measures;
	}

	public List<Integer> getMeasures(){
		return measures;
	}

	public void setName(String name){
		this.name = name;
	}

	public String getName(){
		return name;
	}

	public void setDescription(String description){
		this.description = description;
	}

	public String getDescription(){
		return description;
	}

	public void setId(int id){
		this.id = id;
	}

	public int getId(){
		return id;
	}
}

File: Application/app/src/main/java/com/example/sage/foodsharebook/models/Ingredient.kt
package com.example.sage.foodsharebook.models

import com.google.gson.annotations.SerializedName


class Ingredient(@field:SerializedName("name")
                 var name: String?, @field:SerializedName("description")
                 var description: String?, @field:SerializedName("measures")
                 var measures: List<Int>?) {

    @SerializedName("image")
    var image: String? = null

    @SerializedName("id")
    var id: Int = 0

    val measuresString: String
        get() {
            var result = ""
            for (m in this.measures!!) {
                result += "$m,"
            }
            return result
        }
}